### PR TITLE
Fix Animation 3D Sound

### PR DESCRIPTION
**Pull request by BMagnu**
Previously, all animation sounds were played as 2D sounds.
Now, if they're properly attached to a parent submodel providing position (and they're on an actual object-model as opposed to stuff like skyboxes or cockpits without a position), the sound will play as a proper 3D sound

### CODE REVIEW

**Comment by wookieejedi** [comment]
Tested and works great!

File: code/model/modelanimation_segments.cpp [hunk 1]
```diff
@@ -1213,37 +1217,73 @@ namespace animation {
...
+		if (snd_is_playing(m_instances[pmi->id].currentlyPlaying))
+			snd_stop(m_instances[pmi->id].currentlyPlaying);
 
 		if (m_end.isValid()) 
-			m_instances[pmi_id].currentlyPlaying = snd_play(gamesnd_get_game_sound(m_end));
+			m_instances[pmi->id].currentlyPlaying = playSnd(pmi, m_end, false);
+	}
+
+	sound_handle ModelAnimationSegmentSoundDuring::playSnd(polymodel_instance* pmi, const gamesnd_id& sound, bool loop) {
+		if (m_submodel != nullptr && pmi->objnum > 0) {
```
Comment (Goober5000) [inline code/model/modelanimation_segments.cpp:1236]: Should this be >= ?

Comment (BMagnu) [reply code/model/modelanimation_segments.cpp:1236]: As per object.h:
``int Object_next_signature = 1;	//0 is bogus, start at 1``
Similarly, the scripting API also considers objnum == 0 to be invalid.

Hence, I'd say that checking against > 0 in correct, but don't ask my why it was made that way.

Comment (BMagnu) [reply code/model/modelanimation_segments.cpp:1236]: right. Object _signatures_ are always > 0, but objnums can be 0, I think

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**